### DISCUSSION TITLE
Feature for adding variant specific images

### DISCUSSION THREAD

**Discussion by Bhuvi100** [medusajs/medusa#4456] [Feature Requests]
In most popular ecommerce applications, there will be a lot of variants of the same product and when you choose the desired the variant, the image of the specific variant is shown.

For example say there is a t-shirt with 4 color variants. When you choose the color you need, the t-shirt with the same color you chose will be displayed.

I couldn't find this feature in the product section nor the variants section. It would be helpful because it's a more practical and essential requirement and it won't take a lot of time to implement since there's already a well built image management system present.

**Comment by aihdeveloper** [comment]
Really looking forward to this feature.

**Comment by tsvetann** [comment]
Someone has already developed a plugin for this. Please search in discord

**Reply by yashpratap09** [reply]
please give me link and how to add in our medusa

**Comment by jono-allen** [comment]
https://github.com/andriinuts/medusa-plugin-variant-images/tree/master

**Reply by alsherif-khalaf** [reply]
> https://github.com/andriinuts/medusa-plugin-variant-images/tree/master

This repository has been archived by the owner on Mar 4, 2024. It is now read-only.

**Comment by microCloudCode** [comment]
Really looking forward to this feature.

**Comment by YaminNather** [comment]
Should probably consider making this an inbuilt feature since its common in a ecommerce stores.

**Comment by devtony01** [comment]
Yeah but how can we use it in the storefront

**Comment by alsherif-khalaf** [comment]
it is crucial to have the ability to add Variants image easily

![image](https://github.com/user-attachments/assets/57e97e2e-fbba-41b1-a839-67d92e9920ff)

**Reply by yashpratap09** [reply]
how to do this

**Comment by hijak** [comment]
ah this was a total blocker for me moving from prestashop

**Comment by docloulou** [comment]
Not planned in Medusa 2.0 neither ?

**Comment by joekendal** [comment]
@jono-allen 

> https://github.com/andriinuts/medusa-plugin-variant-images/tree/master

This guy has abandoned it

> As Medusa is not stable and looks like the Medusa team doesn't care about bug fixing I'm archiving this repo. Feel free to fork it if you want. But I would recommend you to have a look at [Vendure](https://vendure.io/) as it's more stable and this functionality is out of the box.

**Comment by joekendal** [comment]
https://github.com/medusajs/medusa/discussions/9444#discussioncomment-11038820

**Comment by Ksztof** [comment]
Here is the solution for medusa v2 created by @Betanoir - https://github.com/Betanoir/medusa-variant-images

**Comment by jaydindg** [comment]
https://github.com/jaydindg/medusajs-variant-images

**Reply by teodora-siljanoska** [reply]
@jaydindg How to configure and install it, please provide me some guidline. Thanks :)

**Comment by codeplott** [comment]
After all this time, it’s happening 🥹!
https://github.com/medusajs/medusa/pull/13670

\
\
Edit: They shipped it in Version [v2.11.2](https://github.com/medusajs/medusa/releases/tag/v2.11.2) 🚀
